virocon/distributions.py

```diff
@@ -1184,3 +1185,163 @@ def _wlsq_error(delta, x, p, w, return_alpha_beta=False, falpha=None, fbeta=None
         wlsq_error = np.sum(w * (x - x_hat) ** 2)
 
         return wlsq_error
+
+
+class GeneralizedGammaDistribution(Distribution):
+    """
+    A generalized Gamma distribution. 
+   
+    The distributions probability density function is given by [5]_ . 
+    The parametrization is orientated on [5]_ :
+    
+    :math:`f(x) = \\frac{ \\lambda^{cm} cx^{cm-1} \\exp \\left[ - \\left(\\lambda x^{c} \\right) \\right] }{\\Gamma(m)}`
+    
+    Parameters
+    ----------
+    m : float
+        Shape parameter of the generalized Gamma distribution. Defaults to 1.
+    c : float
+        Second shape parameter of the generalized Gamma distribution. Defaults
+        to 1.
+    lambda_ : float
+        Additional scale parameter of the generalized Gamma distribution 
+        (3-parameter generalized Gamma distribution). Defaults to 1.
+    f_m : float
+        Fixed shape parameter of the generalized Gamma distribution (e.g. 
+        given physical parameter). If this parameter is set, c is ignored. 
+        Defaults to None.
+    f_c : float
+       Fixed second shape parameter of the generalized Gamma distribution (e.g.
+       given  physical parameter). If this parameter is set, k is ignored. 
+       Defaults to None. 
+    f_lambda_ : float
+        Fixed scale parameter of the generalized Gamma distribution (e.g. 
+        given physical parameter). If this parameter is set, b is ignored. 
+        Defaults to None.
+
+    References
+    ----------
+    .. [5] M.K. Ochi, J.E. Wahlen, Prediction of severest significant wave 
+        height, Coast. Eng. Chap. 36 (1980) 587e599.
+        
+    """
+
+    def __init__(self, m=1, c=1, lambda_=1, f_m=None, f_c=None, f_lambda_=None):
+
+        # TODO set parameters to fixed values if provided
```

Comment:
Is this TODO done? then remove the comment.

After briefly scanning this TODO seems not done. You could test this, if you instantiate a new distribution with non-default f_params and see which value is used e.g. for the pdf.(Without calling fit)

Reply:
Actually I think this TODO is done, at least after I tested what you suggested. What do you think?

<img width="549" alt="Bildschirmfoto 2021-11-22 um 11 55 00" src="https://user-images.githubusercontent.com/57865209/142849278-4e3ac769-e25e-4ce7-899f-c29379aac0b1.png">

Reply:
The test does not what I meant. This only shows that the f_x are set when supplied to the constructor. But what we really want to know is, what is the value of m if f_m is set.